lib/ood_core/job/cluster_info.rb

```diff
@@ -0,0 +1,31 @@
+module OodCore
+  module Job
+    # An object that contains details about the cluster's active and total nodes, processors, and gpu nodes
+    class ClusterInfo
+      attr_reader :cluster, :nodes_active, :nodes_total, :processors_active, :processors_total, :gpu_nodes_active,
+                  :gpu_nodes_total
+
+      def initialize(cluster_name: nil, nodes_active: nil, nodes_total: nil, processors_active: nil, processors_total: nil,
+                     gpu_nodes_active: nil, gpu_nodes_total: nil, **_)
+        @cluster_name        = cluster
+        @nodes_active        = nodes_active
+        @nodes_total         = nodes_total
+        @processors_active   = processors_active
+        @processors_total    = processors_total
+        @gpu_nodes_active    = gpu_nodes_active
+        @gpu_nodes_total     = gpu_nodes_total
+      end
```

Comment:
I didn't want to type all these out, so here's just the one that you can repeat. (I also reversed the adjective + noun in the name).

Also  - you don't need cluster (or `cluster_name`). If I'm doing `clusters[:owens].job_adapter.cluster_info` I already know that it's the `owens` cluster.

```suggestion
      def initialize(opts = {})
        opts = opts.symbolize_keys
        @active_nodes        = opts.fetch(:active_nodes, nil).to_i
      end
```